### PR TITLE
fix(builtin): require correct runfiles path to chdir script

### PR DESCRIPTION
**Pull request by kormide**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature (please, look at the "Scope of the project" section in the README.md file)
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

`chdir` with `nodejs_binary` is currently broken because for some reason `rlocation` does not return the path to the generated chdir script in the runfiles dir, but instead in the output dir. This only happens under a `bazel run`, not when running `test` on a `nodejs_test`, so I added an e2e test to cover this case.

## What is the new behavior?

We now try to load the chdir script from the `$RUNFILES` path if it exists, falling back to rlocation if it doesn't.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: internal/node/node.bzl [hunk 1]
```diff
@@ -322,6 +322,8 @@ if (process.cwd() !== __dirname) {
         "TEMPLATED_runfiles_helper_script": _to_manifest_path(ctx, ctx.file._runfile_helpers_main),
         "TEMPLATED_node_tool_path": strip_external(node_toolchain.nodeinfo.target_tool_path),
         "TEMPLATED_node_args": ctx.attr._node_args[UserBuildSettingInfo].value,
+        "TEMPLATED_chdir_runfiles_path": chdir_script_runfiles_path if ctx.attr.chdir else "*not used*",
+        "TEMPLATED_chdir": "true" if ctx.attr.chdir else "false",
```
Comment (kormide) [inline internal/node/node.bzl:326]: Hm, I can probably simplify this into a single template variable.

File: e2e/bazel_run_chdir/WORKSPACE [hunk 2]
```diff
@@ -0,0 +1,17 @@
+workspace(name = "e2e_nodejs_host")
```
Comment (gregmagolan) [inline e2e/bazel_run_chdir/WORKSPACE:1]: nit: update workspace name to match the folder name

**Review by gregmagolan — APPROVED** [review]
Beauty. Just the Windows failure to sort out but looks great otherwise.

**Comment by kormide** [comment]
Added bazelrc to e2e test and disabled windows ci run, as discussed.

**Review by alexeagle — APPROVED** [review]

**Pull request merged**